OS X build fails due to missing elf.h

```
make[2]: Entering directory '/Users/filippo/build/musl-cross-make/linux-4.4.10'
make[3]: Entering directory '/Users/filippo/build/musl-cross-make/build-x86_64-linux-musl/obj_kernel_headers'
  CHK     include/generated/uapi/linux/version.h
  UPD     include/generated/uapi/linux/version.h
  HOSTCC  scripts/basic/fixdep
  WRAP    arch/x86/include/generated/asm/clkdev.h
  WRAP    arch/x86/include/generated/asm/cputime.h
  WRAP    arch/x86/include/generated/asm/dma-contiguous.h
  WRAP    arch/x86/include/generated/asm/early_ioremap.h
  WRAP    arch/x86/include/generated/asm/mcs_spinlock.h
  WRAP    arch/x86/include/generated/asm/mm-arch-hooks.h
  SYSTBL  arch/x86/entry/syscalls/../../include/generated/asm/syscalls_32.h
  SYSHDR  arch/x86/entry/syscalls/../../include/generated/uapi/asm/unistd_32.h
  SYSHDR  arch/x86/entry/syscalls/../../include/generated/uapi/asm/unistd_64.h
  SYSHDR  arch/x86/entry/syscalls/../../include/generated/uapi/asm/unistd_x32.h
  HOSTCC  arch/x86/tools/relocs_32.o
In file included from /Users/filippo/build/musl-cross-make/linux-4.4.10/arch/x86/tools/relocs_32.c:1:
/Users/filippo/build/musl-cross-make/linux-4.4.10/arch/x86/tools/relocs.h:12:10: fatal error: 'elf.h' file not found
#include <elf.h>
         ^
1 error generated.
scripts/Makefile.host:108: recipe for target 'arch/x86/tools/relocs_32.o' failed
make[4]: *** [arch/x86/tools/relocs_32.o] Error 1
arch/x86/Makefile:193: recipe for target 'archscripts' failed
make[3]: *** [archscripts] Error 2
make[3]: Leaving directory '/Users/filippo/build/musl-cross-make/build-x86_64-linux-musl/obj_kernel_headers'
Makefile:146: recipe for target 'sub-make' failed
make[2]: *** [sub-make] Error 2
make[2]: Leaving directory '/Users/filippo/build/musl-cross-make/linux-4.4.10'
Makefile:196: recipe for target 'obj_kernel_headers/.lc_built' failed
make[1]: *** [obj_kernel_headers/.lc_built] Error 2
make[1]: Leaving directory '/Users/filippo/build/musl-cross-make/build-x86_64-linux-musl'
Makefile:152: recipe for target 'all' failed
make: *** [all] Error 2
```

```
TARGET = x86_64-linux-musl
OUTPUT = /usr/local/Cellar/musl-cross-make/HEAD
```

--- a/patches/binutils-2.25.1/0007-tcarm.diff
+++ b/patches/binutils-2.25.1/0007-tcarm.diff
@@ -1,0 +1,21 @@
+From d840c081f8082e8b9e63fead5306643975a97bb3 Mon Sep 17 00:00:00 2001
+From: Richard Earnshaw <Richard.Earnshaw@arm.com>
+Date: Thu, 20 Nov 2014 17:02:47 +0000
+Subject: [PATCH] * config/tc-arm.c (rotate_left): Avoid undefined behaviour
+ when N = 0.
+
+---
+
+diff --git a/gas/config/tc-arm.c b/gas/config/tc-arm.c
+index 5077f87..9100fb2 100644
+--- a/gas/config/tc-arm.c
++++ b/gas/config/tc-arm.c
+@@ -7251,7 +7251,7 @@ parse_operands (char *str, const unsigned int *pattern, bfd_boolean thumb)
+ 
+ /* Functions for operand encoding.  ARM, then Thumb.  */
+ 
+-#define rotate_left(v, n) (v << n | v >> (32 - n))
++#define rotate_left(v, n) (v << (n & 31) | v >> ((32 - n) & 31))
+ 
+ /* If VAL can be encoded in the immediate field of an ARM instruction,
+    return the encoded form.  Otherwise, return FAIL.  */

--- a/patches/linux-4.4.10/0001-archscripts.diff
+++ b/patches/linux-4.4.10/0001-archscripts.diff
@@ -1,0 +1,12 @@
+diff -u a/Makefile b/Makefile
+--- a/Makefile	2016-05-11 10:23:26.000000000 +0100
++++ b/Makefile	2016-05-20 16:26:20.000000000 +0100
+@@ -1047,7 +1047,7 @@
+ archscripts:
+ 
+ PHONY += __headers
+-__headers: $(version_h) scripts_basic asm-generic archheaders archscripts FORCE
++__headers: $(version_h) scripts_basic asm-generic archheaders FORCE
+ 	$(Q)$(MAKE) $(build)=scripts build_unifdef
+ 
+ PHONY += headers_install_all

